crates/optimism/payload/src/builder.rs

```diff
@@ -684,9 +682,14 @@ where
         Builder: BlockBuilder<Primitives = Evm::Primitives>,
         <<Builder::Executor as BlockExecutor>::Evm as AlloyEvm>::DB: Database,
     {
-        let block_gas_limit = builder.evm_mut().block().gas_limit();
-        let block_da_limit = self.da_config.max_da_block_size();
-        let tx_da_limit = self.da_config.max_da_tx_size();
+        let mut block_gas_limit = builder.evm_mut().block().gas_limit();
+        if let Some(gas_limit_config) = self.builder_config.gas_limit_config.gas_limit() {
+            // If a gas limit is configured, use that limit as target if it's smaller, otherwise use
+            // the block's actual gas limit.
+            block_gas_limit = gas_limit_config.min(block_gas_limit);
+        };
```

Comment:
@haardikk21 this is the actual integration for the sequencer tx execution.

this doesnt alter the block's gas limit but enforces how much we fill the block with

Reply:
Perfect